03_Modules/Certificates/src/module/module.ts

```diff
@@ -193,94 +166,52 @@ export class CertificatesModule extends AbstractModule {
     props?: HandlerProperties,
   ): Promise<void> {
     this._logger.debug('Sign certificate request received:', message, props);
+    const stationId: string = message.context.stationId;
+    // when parsing pem string, node forge expect a csr
+    // which has header and footer without new line characters.
+    const csrString: string = message.payload.csr.replace(/\n/g, '');
+    const certificateType: CertificateSigningUseEnumType | undefined =
+      message.payload.certificateType;
+
     // OCTT Currently fails the CSMS on test case TC_A_14_CSMS if an invalid csr is rejected
     // Despite explicitly saying in the protocol "The CSMS may do some checks on the CSR"
     // So it is necessary to accept before checking the csr. when this is fixed, this line can be removed
     // And the other sendCallResultWithMessage for SignCertificateResponse can be uncommented
     this.sendCallResultWithMessage(message, {
       status: GenericStatusEnumType.Accepted,
     } as SignCertificateResponse);
-    let caCert: forge.pki.Certificate;
-    let caPrivateKey: forge.pki.rsa.PrivateKey;
-    let csr;
+
+    // Do Verification and send accept response before signing.
+    // Reject the request if the verification fails
     try {
-      csr = forge.pki.certificationRequestFromPem(message.payload.csr);
+      await this._verifySignCertRequest(csrString, stationId, certificateType);
+      // this.sendCallResultWithMessage(message, {
```

Comment:
is this commented out temporarily? wondering if an additional TODO comment to be like "will be uncommented in another PR" is needed so that it's not confusing why there's commented out code 😅